src/main/java/com/example/bookstoreapp/services/impl/CartItemServiceImpl.java

```diff
@@ -12,159 +12,164 @@
+        CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+        CriteriaQuery<CartItemEntity> criteriaQuery = criteriaBuilder.createQuery(CartItemEntity.class);
+        Root<CartItemEntity> root = criteriaQuery.from(CartItemEntity.class);
 
-  @Autowired
-  private EntityManager entityManager;
+        // Adding a predicate to the query to filter by catalogItemId
+        Predicate predicate1 = criteriaBuilder.equal(root.get("catalogItemId"), catalogItemId);
+        Predicate predicate2 = criteriaBuilder.equal(root.get("cartId"), userContextService.getUserId());
+        criteriaQuery.where(predicate1, predicate2);
 
-  @Autowired
-  private CartItemEntityRepository cartItemEntityRepository;
+        List<CartItemEntity> resultList = entityManager.createQuery(criteriaQuery).setMaxResults(1).getResultList();
 
-  @Override
-  public CartItem addItem(Long catalogItemId) {
-    CatalogItem catalogItem = catalogItemService.findById(catalogItemId);
-    CartItemEntity savedCartItemEntity = findCartItemEntity(catalogItemId);
-    if(catalogItem.getStockQuantity() <= 0){
-      throw new AppRuntimeException("Item is out of stock");
+        return resultList.isEmpty() ? null : resultList.get(0);
     }
-    if (savedCartItemEntity == null) { // create new entity
-      CartItemEntity newCartItemEntity = new CartItemEntity();
-      newCartItemEntity.setId(IdGenerator.getLongId());
-      newCartItemEntity.setCartId(userContextService.getUserId());
-      newCartItemEntity.setQuantity(1);
-      newCartItemEntity.setCatalogItemId(catalogItemId);
-      newCartItemEntity.setState(CartItemState.BUY_NOW.getWeight());
-
-      newCartItemEntity = cartItemEntityRepository.save(newCartItemEntity);
 
-      return new CartItem().fromEntity(newCartItemEntity, catalogItem);
+    private List<CartItemEntity> findCartItemEntities(CartItemState cartItemState) {
+        // fetch from db using criteria query
+        CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+        CriteriaQuery<CartItemEntity> criteriaQuery = criteriaBuilder.createQuery(CartItemEntity.class);
+        Root<CartItemEntity> root = criteriaQuery.from(CartItemEntity.class);
+
+        List<Predicate> predicateList = new ArrayList<>();
+        // Adding a predicate to the query to filter by cartId
+        Predicate cartIdPredicate = criteriaBuilder.equal(root.get("cartId"), userContextService.getUserId());
+        predicateList.add(cartIdPredicate);
+        if (cartItemState != null) {
+            Predicate statePredicate = criteriaBuilder.equal(root.get("state"), cartItemState.getWeight());
+            predicateList.add(statePredicate);
+        }
+        criteriaQuery.where(predicateList.toArray(new Predicate[0]));
+        // select * from cart_item where cart_id=?
+        // select * from cart_item where cart_id=? and state=?
+
+        return entityManager.createQuery(criteriaQuery).getResultList();
     }
-    // increase the quantity
```

Comment:
why does it looks like below lines cleaned up?

Reply:
yes I have added code clean up